core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -96,6 +96,11 @@
         Dynamic,
 
         /**
+         * mark this setting as final (the value cannot be updated)
```

Comment:
> I think it is worth explaining here how this is different than not marking the setting as Dynamic.

The value cannot be updated in non-dynamic context ? ;)
My understanding is that `Dynamic` means a setting that can be updated dynamically.
This does not prevent this setting to be updated non-dynamically so I think that `Final` makes sense as an abstract property in the setting object. Non dynamic contexts are defined at the scope level, so for instance updating the settings on a closed index is considered as non-dynamic. And if we want to add more non-dynamic context we should just ensure that this property is applied ?